### PR TITLE
Tj/stax porting

### PR DESCRIPTION
**Pull request by tdejoigny-ledger**
- add Stax device support (including makefile update, icons, glyphs, Stax UI flows)
- securize/factorize the crypto functions
- update UI for nanos (same UI as NanoSP/X)
- remove ledgerBlue support
- fix last CI issues

### CODE REVIEW

File: src/main.h
Comment (xchapron-ledger) [inline src/main.h:28]:
```suggestion
#endif

```
Missing new line at the end of the file.

File: src/ui/ui_nbgl.c
Comment (xchapron-ledger) [inline src/ui/ui_nbgl.c:373]:
```suggestion
```

File: src/ui/ui_nbgl.h
Comment (xchapron-ledger) [inline src/ui/ui_nbgl.h:39]:
```suggestion
#endif

```

File: glyphs/icon_warning_64px.gif
Comment (tdejoigny-ledger) [reply glyphs/icon_warning_64px.gif:1]: In theory, yes. But at term this icon will have to be done by a designer in 4bpp and included in SDK.

File: Makefile
Comment (tdejoigny-ledger) [reply Makefile:28]: yes I checked on real devices (Stax and Nano SP). No issue found.

File: src/ui/ui_nbgl.c
Comment (tdejoigny-ledger) [reply src/ui/ui_nbgl.c:373]: ok

File: src/ui/ui_nbgl.h
Comment (tdejoigny-ledger) [reply src/ui/ui_nbgl.h:39]: ok

File: src/main.h
Comment (tdejoigny-ledger) [reply src/main.h:28]: ok

File: README.md
Comment (tdejoigny-ledger) [reply README.md:2]: ok

File: glyphs/icon_message_64px.gif
Comment (xchapron-ledger) [reply glyphs/icon_message_64px.gif:1]: So why not putting them there right now? So that when a designer changed them to 4bpp it would be change on the app automatically?
And it will make it sure that they are not forgotten by the designer and correctly shared between all apps?

**Review by xchapron-ledger — APPROVED** [review]

File: common/vetDisplay.c
Comment (tdejoigny-ledger) [reply common/vetDisplay.c:77]: ok

File: src/ui/ui_nbgl.c
Comment (tdejoigny-ledger) [reply src/ui/ui_nbgl.c:92]: ok

**Comment by tdejoigny-ledger** [comment]
Hello @libotony, 

I will merge today if you don't have any extra comments.

**Comment by libotony** [comment]
> Hello @libotony,
> 
> I will merge today if you don't have any extra comments.

Merge this or #13 ?

**Comment by libotony** [comment]
I haven't finished all files, I will add some comments today, could you hold it?

**Comment by tdejoigny-ledger** [comment]
ok I will wait your last comments....

File: src/main.c
Comment (libotony) [inline src/main.c:116]: The signature length is 65, can this be set to `signature[65]`?

File: src/main.c
Comment (libotony) [inline src/main.c:577]: Should be `CX_BLAKE2B`

**Comment by libotony** [comment]
When do you plan to release this after this PR is merged? I have finished the review of `main.c` but still need some time to work on stax's UX test flow.

**Comment by tdejoigny-ledger** [comment]
@libotony Do you have an account on discord ? it will be easier to discuss.

**Comment by libotony** [comment]
> @libotony Do you have an account on discord ? it will be easier to discuss.

Yes, libotony#0900

File: src/main.c
Comment (tdejoigny-ledger) [reply src/main.c:577]: not sure. It is not in the scope of Stax porting. As discussed, you could open a subsequently PR to address this point if it is required.

File: src/main.c
Comment (tdejoigny-ledger) [reply src/main.c:1011]: It is not in the scope of Stax porting. As discussed, you could open a subsequently PR to address this point if it is required.

**Review by xchapron-ledger — APPROVED** [review]

**Pull request merged**
